telos-backend/src/models/text.js

```diff
@@ -0,0 +1,16 @@
+const mongoose = require('mongoose');
+
+const textSchema = new mongoose.Schema({
+  text: {
+    type: String,
+    required: true,
+  },
+  // foreign-key to widget model
+  widgetId: {
```

Comment:
I think by default this won't perform any validation, ie its just storing a uuid, its not verifying that theres some document with this ObjectId in the database. We need to be careful if we assume this works like a foreign key. It looks like we might require some middleware eg. https://www.npmjs.com/package/mongoose-id-validator. But I think we can assume that ID's will be passed around in a safe way for the MVP.

Reply:
Yup, that makes sense, this was another thing I looked a lot into since it was between either storing just the string or the whole ObjectID. When searching up MongoDB foreign keys though, it all pointed to doing it this way. 

This is an area we will need to test to see if it can be transferred between backend to frontend while preserving its ObjectID structure. If that proves to be an issue, we can go back to just using the string value.

Reply:
I still think storing ObjectId is the right call, but just that its possible that the frontend could pass a value that doesn't match any document in the database. But I don't think it should be too much of an issue, just something I thought I should point out in case it becomes a problem later.